src/state.rs

```diff
@@ -357,34 +560,26 @@ impl DemandProxyState {
     }
 }
 
-fn set_gateway_address(us: &mut Upstream, hbone_port: u16) -> anyhow::Result<()> {
+pub fn set_gateway_address(
```

Comment:
maybe a nit, but why does this take a `mut Upstream` and not exist on `impl Upstream`?